hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/rollback/ListingBasedRollbackStrategy.java

```diff
@@ -57,20 +84,242 @@ public ListingBasedRollbackStrategy(HoodieTable table,
+        String fileCommitTime = FSUtils.getCommitTime(path.getName());
+        return commit.equals(fileCommitTime);
+      } else if (FSUtils.isLogFile(path)) {
+        // Since the baseCommitTime is the only commit for new log files, it's okay here
+        String fileCommitTime = FSUtils.getBaseCommitTimeFromLogPath(path);
+        return commit.equals(fileCommitTime);
+      }
+      return false;
+    };
+  }
+
+  private static List<ListingRollbackActionType> getListingRollbackType(HoodieInstant instantToRollback,
+                                                                        HoodieTable<?, ?, ?, ?> table, String partitionPath) throws IOException {
+    List<ListingRollbackActionType> listingRollbackActionTypes = new ArrayList<>(2);
+    if (table.getMetaClient().getTableType() == HoodieTableType.COPY_ON_WRITE) {
+      listingRollbackActionTypes.add(ListingRollbackActionType.DELETE_DATA_AND_LOG_FILES);
+    } else {
+      String commit = instantToRollback.getTimestamp();
+      HoodieActiveTimeline activeTimeline = table.getMetaClient().reloadActiveTimeline();
+      switch (instantToRollback.getAction()) {
+        case HoodieTimeline.COMMIT_ACTION:
+        case HoodieTimeline.REPLACE_COMMIT_ACTION:
+          listingRollbackActionTypes.add(ListingRollbackActionType.DELETE_DATA_AND_LOG_FILES);
+          break;
+        case HoodieTimeline.COMPACTION_ACTION:
+          // If there is no delta commit present after the current commit (if compaction), no action, else we
+          // need to make sure that a compaction commit rollback also deletes any log files written as part of the
+          // succeeding deltacommit.
+          boolean higherDeltaCommits =
+              !activeTimeline.getDeltaCommitTimeline().filterCompletedInstants().findInstantsAfter(commit, 1).empty();
+          if (higherDeltaCommits) {
+            // Rollback of a compaction action with no higher deltacommit means that the compaction is scheduled
+            // and has not yet finished. In this scenario we should delete only the newly created base files
+            // and not corresponding base commit log files created with this as baseCommit since updates would
+            // have been written to the log files.
+            listingRollbackActionTypes.add(ListingRollbackActionType.DELETE_DATA_FILES_ONLY);
+          } else {
+            // No deltacommits present after this compaction commit (inflight or requested). In this case, we
+            // can also delete any log files that were created with this compaction commit as base
+            // commit.
+            listingRollbackActionTypes.add(ListingRollbackActionType.DELETE_DATA_AND_LOG_FILES);
+          }
+          break;
+        case HoodieTimeline.DELTA_COMMIT_ACTION:
+          // --------------------------------------------------------------------------------------------------
+          // (A) The following cases are possible if index.canIndexLogFiles and/or index.isGlobal
```

Comment:
Note to self: we need to review and update this comments as they seem to be out of sync with the code